### PR TITLE
Add paho-mqtt-cpp/1.1

### PR DESCRIPTION
**Pull request by chhitz**
Specify library name and version:  **paho-mqtt-cpp/1.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by chhitz** [comment]
From https://github.com/conan-community/conan-paho-cpp/pull/7

File: recipes/paho-mqtt-cpp/all/conanfile.py [hunk 1]
```diff
@@ -65,8 +66,9 @@ def _configure_cmake(self):
         return self._cmake
 
     def build(self):
-        for patch in self.conan_data["patches"][self.version]:
-            tools.patch(**patch)
+        if "patches" in self.conan_data.keys():
```
Comment (uilianries) [inline recipes/paho-mqtt-cpp/all/conanfile.py]:
```suggestion
        if self.version in self.conan_data["patches"]:
```

patches will be present due the previous version, however, the patch version may vary.

Comment (chhitz) [reply recipes/paho-mqtt-cpp/all/conanfile.py]: This was actually my first try, but due to a previous:
> [HOOK - conan-center.py] post_export(): [CONANDATA.YML REDUCE (KB-H031)] Saving conandata.yml: {'sources': {'1.1': {'sha256': 'cb0343349ed91ef51d0e76ae860d19435a730d3d355e57886bb090014cb70bbe', 'url': 'https://github.com/eclipse/paho.mqtt.cpp/archive/v1.1.tar.gz'}}}

there is actually no "patches" in `conan_data` left.

Comment (Croydon) [reply recipes/paho-mqtt-cpp/all/conanfile.py]: I run into that previously as well. I believe that the shrinking of the `conandata.yml` file leaves no `patches` key behind if there are no patch files for the current version

**Comment by chhitz** [comment]
Closing in favour of https://github.com/conan-io/conan-center-index/pull/1099

**Pull request closed without merging**